[fix] Node Engine Requirement?

**ReactUSWDS Version & USWDS Version:**
ReactUSWDS: v2.0.0
USWDS: n/a

<!-- Please note what versions of both ReactUSWDS and USWDS this bug pertains to. -->

**Describe the bug**
Unable to install when using a node engine >14. Unsure if this is a bug or an undocumented requirement.

<!-- A clear and concise description of what the bug is. -->

**To Reproduce**
Steps to reproduce the behavior:
`npm i @trussworks/react-uswds` with node@16.5.0 (LTS changes to v16 on 2021-10-26)

out:
```bash
❯ npm i @trussworks/react-uswds
npm ERR! code EBADENGINE
npm ERR! engine Unsupported engine
npm ERR! engine Not compatible with your version of node/npm: @trussworks/react-uswds@2.0.0
npm ERR! notsup Not compatible with your version of node/npm: @trussworks/react-uswds@2.0.0
npm ERR! notsup Required: {"node":"10.x - 14.x"}
npm ERR! notsup Actual:   {"npm":"7.19.1","node":"v16.5.0"}
```

**Expected behavior**
Successful install




_Thanks for putting together this library, I'm excited to try it out 😄_

--- a/.github/workflows/build-and-test.yml
+++ b/.github/workflows/build-and-test.yml
@@ -10,6 +10,12 @@ jobs:
   build-and-test:
     runs-on: ubuntu-latest
 
+    strategy:
+      matrix:
+        node: ['12', '14', '16']
+
+    name: Node ${{ matrix.node }} tests
+
     steps:
       - name: Check out repository
         uses: actions/checkout@v2
@@ -17,10 +23,9 @@ jobs:
       - name: Set up Node.js
         uses: actions/setup-node@v2
         with:
-          node-version: 12.x
+          node-version: ${{ matrix.node }}
           cache: 'yarn'
 
-      # The yarn cache is not node_modules
       - name: Install dependencies
         run: yarn --frozen-lockfile
 

--- a/.github/workflows/build-and-test.yml
+++ b/.github/workflows/build-and-test.yml
@@ -40,6 +40,12 @@ jobs:
       - name: Check out repository
         uses: actions/checkout@v2
 
+      - name: Set up Node.js
+        uses: actions/setup-node@v2
+        with:
+          node-version: 12.x
+          cache: 'yarn'
+
       - name: Happo
         run: yarn happo-ci
         env:

--- a/.github/workflows/build-and-test.yml
+++ b/.github/workflows/build-and-test.yml
@@ -30,9 +30,6 @@ jobs:
       - name: Run node server side tests
         run: yarn test:serverside
 
-      - name: Lint code
-        run: yarn lint
-
   run-checks:
     runs-on: ubuntu-latest
 
@@ -46,6 +43,13 @@ jobs:
           node-version: 12.x
           cache: 'yarn'
 
+      # The yarn cache is not node_modules
+      - name: Install dependencies
+        run: yarn --frozen-lockfile
+
+      - name: Lint code
+        run: yarn lint
+
       - name: Happo
         run: yarn happo-ci
         env:
@@ -53,6 +57,6 @@ jobs:
           HAPPO_API_SECRET: ${{ secrets.HAPPO_API_SECRET }}
 
       - name: Run Danger JS
-        run: danger/danger-js@9.1.8
+        run: yarn danger ci --failOnErrors
         env:
           GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}

--- a/.github/workflows/build-and-test.yml
+++ b/.github/workflows/build-and-test.yml
@@ -16,7 +16,7 @@ jobs:
 
       - name: Set up Node.js
         uses: actions/setup-node@v2
-        with: 
+        with:
           node-version: 12.x
           cache: 'yarn'
 
@@ -33,13 +33,20 @@ jobs:
       - name: Lint code
         run: yarn lint
 
+  run-checks:
+    runs-on: ubuntu-latest
+
+    steps:
+      - name: Check out repository
+        uses: actions/checkout@v2
+
       - name: Happo
         run: yarn happo-ci
-        env: 
+        env:
           HAPPO_API_KEY: ${{ secrets.HAPPO_API_KEY }}
           HAPPO_API_SECRET: ${{ secrets.HAPPO_API_SECRET }}
 
-      - name: Run Danger JS 
-        run: yarn danger ci --failOnErrors
-        env: 
+      - name: Run Danger JS
+        run: danger/danger-js@9.1.8
+        env:
           GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}

--- a/package.json
+++ b/package.json
@@ -17,7 +17,7 @@
     "lib"
   ],
   "engines": {
-    "node": "10.x - 14.x"
+    "node": "10.x - 16.x"
   },
   "scripts": {
     "test": "jest",

